[frontend] Fix carousel alignment issues

clairezzhang opened the pull request:
After merging #567 into dev, I noticed that there were alignment issues with how the carousel fit into the homepage. I fixed these alignment issues and made a new css file (static/carousel.css) for the carousel code so that it would be easier to find than in its previous location (static/machina/custom-styling.css).

<img width="1462" alt="image" src="https://github.com/user-attachments/assets/0a7ba2bd-9be4-4fec-9d86-8c3077c008ef" />

Fixes: #567 and #574

bluto0 committed:
creating carousel

bluto0 committed:
Merge branch 'dev' into carousel

bluto0 committed:
slight changes

bluto0 committed:
fixed layout bug

bluto0 committed:
Merge branch 'carousel' of https://github.com/uchicago-cs/chigame into carousel

clairezzhang committed:
Merge branch 'dev' into carousel

rtvazquez committed:
Merge branch 'dev' into carousel

rtvazquez committed:
Merge branch 'dev' into carousel

rtvazquez committed:
Merge branch 'dev' into carousel

clairezzhang committed:
[frontend] fix game carousel alignment post merge

Fixed alignment issues that game carousel had when merged with
homepage code. Also moved the carousel css to a new, more aptly
named carousel.css file in the static/css folder.

Fixes: #567

clairezzhang committed:
Merge branch 'dev' into carousel

pjacobsen1 reviewed (approved):
The site ran smoothly, and the carousel looks as expected, so seems all set to me!

jacqsmei commented on src/static/css/homepage.css:
```diff
@@ -104,7 +104,7 @@
   border: 1px solid #D9D9D9;
   border-radius: 5px;
-  padding: 0.5rem;
+  padding: 0.5rem 0.5rem 0rem 0.5rem;
```
this is a nitpick, but i don't think this change is needed

clairezzhang replied:
I took a look and I think you're right! I think I changed it when I was trying to figure out why the alignment was off, but it wasn't the reason.

jacqsmei reviewed (changes requested):
the rest looks good to me! nice catch

clairezzhang committed:
Merge branch 'dev' into carousel

clairezzhang committed:
[frontend] Remove unnecessary homepage css change

As mentioned in #757, one change made in the pull request was unnecessary.
I double checked its effect and removed it.

Fixes: #567

clairezzhang committed:
Merge branch 'dev' into carousel

jacqsmei committed:
Merge branch 'dev' into carousel

jacqsmei reviewed (approved):
nice work!

jacqsmei commented:
Issue Score: Excellent

Comments:
Nice work catching this bug! Make sure everyone's commits follow the developer guideline standards!

<!-- {"score": "excellent", "minor": false} -->
